b_pull_Landsat_SRST_poi/py/gee_functions.py

```diff
@@ -0,0 +1,674 @@
+    "Blue": image.select(["Blue"]),
+    "Green": image.select(["Green"]),
+    "mbsrn": Mbsrn(image).select(["mbsrn"]),
+    "Swir2": image.select(["Swir2"])
+  }))
+
+
+## The DSWE Function itself    
+def DSWE(image):
+  """calculate the dynamic surface water extent per pixel
+  
+  Args:
+      image: ee.Image of an ee.ImageCollection
+      
+  Returns:
+      band where values calculated are the DSWE value per pixel
+  """
+  mndwi = Mndwi(image)
+  mbsrv = Mbsrv(image)
+  mbsrn = Mbsrn(image)
+  awesh = Awesh(image)
+  swir1 = image.select(["Swir1"])
+  nir = image.select(["Nir"])
+  ndvi = Ndvi(image)
+  blue = image.select(["Blue"])
+  swir2 = image.select(["Swir2"])
+  # These thresholds are taken from the LS Collection 2 DSWE Data Format Control Book
+  # Inputs are meant to be scaled reflectance values 
+  t1 = mndwi.gt(0.124) # MNDWI greater than Wetness Index Threshold
+  t2 = mbsrv.gt(mbsrn) # MBSRV greater than MBSRN
+  t3 = awesh.gt(0) #AWESH greater than 0
+  t4 = (mndwi.gt(-0.44)  #Partial Surface Water 1 thresholds
+   .And(swir1.lt(0.09)) #900 for no scaling (LS Collection 1)
+   .And(nir.lt(0.15)) #1500 for no scaling (LS Collection 1)
+   .And(ndvi.lt(0.7)))
+  t5 = (mndwi.gt(-0.5) #Partial Surface Water 2 thresholds
+   .And(blue.lt(0.1)) #1000 for no scaling (LS Collection 1)
+   .And(swir1.lt(0.3)) #3000 for no scaling (LS Collection 1)
+   .And(swir2.lt(0.1)) #1000 for no scaling (LS Collection 1)
+   .And(nir.lt(0.25))) #2500 for no scaling (LS Collection 1)
+  t = (t1
+    .add(t2.multiply(10))
+    .add(t3.multiply(100))
+    .add(t4.multiply(1000))
+    .add(t5.multiply(10000)))
+  noWater = (t.eq(0)
+    .Or(t.eq(1))
+    .Or(t.eq(10))
+    .Or(t.eq(100))
+    .Or(t.eq(1000)))
+  hWater = (t.eq(1111)
+    .Or(t.eq(10111))
+    .Or(t.eq(11011))
+    .Or(t.eq(11101))
+    .Or(t.eq(11110))
+    .Or(t.eq(11111)))
+  mWater = (t.eq(111)
+    .Or(t.eq(1011))
+    .Or(t.eq(1101))
+    .Or(t.eq(1110))
+    .Or(t.eq(10011))
+    .Or(t.eq(10101))
+    .Or(t.eq(10110))
+    .Or(t.eq(11001))
+    .Or(t.eq(11010))
+    .Or(t.eq(11100)))
+  pWetland = t.eq(11000)
+  lWater = (t.eq(11)
+    .Or(t.eq(101))
+    .Or(t.eq(110))
+    .Or(t.eq(1001))
+    .Or(t.eq(1010))
+    .Or(t.eq(1100))
+    .Or(t.eq(10000))
+    .Or(t.eq(10001))
+    .Or(t.eq(10010))
+    .Or(t.eq(10100)))
+  iDswe = (noWater.multiply(0)
+    .add(hWater.multiply(1))
+    .add(mWater.multiply(2))
+    .add(pWetland.multiply(3))
+    .add(lWater.multiply(4)))
+  return iDswe.rename("dswe")
+
+
+def calc_hill_shades(image, geo):
+  """ caluclate the hill shade per pixel
+
+  Args:
+      image: ee.Image of an ee.ImageCollection
+      geo: geometry of the WRS tile as wrs.geometry() in script
+
+  Returns:
+      a band named "hillShade" where values calculated are the hill shade per 
+      pixel. output is 0-255. 
+  """
+  MergedDEM = ee.Image("users/eeProject/MERIT").clip(geo.buffer(3000))
```

Comment:
```suggestion
  MergedDEM = ee.Image("MERIT/DEM/v1_0_3").clip(geo.buffer(3000));
```
GEE now has an official asset for the MERIT DEM.